redisinsight/api/src/modules/browser/services/redisearch/redisearch.service.ts

```diff
@@ -147,16 +150,33 @@ export class RedisearchService {
         new Command('FT.SEARCH', [index, query, 'NOCONTENT', 'LIMIT', offset, limit]),
       );
 
+
+      try {
+        const [[, maxSearchResults]] = await client.sendCommand(
+          // response: [ [ 'MAXSEARCHRESULTS', '10000' ] ]
+          new Command('FT.CONFIG', ['GET', 'MAXSEARCHRESULTS'], {
+            replyEncoding: 'utf8',
+          }),
+        ) as [[string, string]];
+
+        maxResults = toNumber(maxSearchResults);
+      } catch (error) {
+        maxResults = null;
+      }
+
       return plainToClass(GetKeysWithDetailsResponse, {
         cursor: limit + offset,
         total,
         scanned: keyNames.length + offset,
         keys: keyNames.map((name) => ({ name })),
+        maxResults,
       });
-    } catch (e) {
-      this.logger.error('Failed to search keys using redisearch index', e);
-
-      throw catchAclError(e);
+    } catch (error) {
+      this.logger.error('Failed to search keys using redisearch index', error);
+      if (error.message?.includes(RedisErrorCodes.RedisearchLimit)) {
+        throw new BadRequestException(ERROR_MESSAGES.INCREASE_MINIMUM_LIMIT(dto.limit?.toLocaleString?.()));
```

Comment:
`ERROR_MESSAGES.INCREASE_MINIMUM_LIMIT(dto.limit?.toLocaleString?.())`
May be `dto.offset` or `dto.limit + dto.offset`? ideally limit will never change across "show more" requests (e.g. always 10 000)

Reply:
Only config for limit throws an error. For the Browser view limit is 500.